pandas/core/arrays/datetimes.py

```diff
@@ -817,21 +817,34 @@ def _add_offset(self, offset: BaseOffset) -> Self:
             result = type(self)._from_sequence(res_values, dtype=self.dtype)
 
         else:
-            units = ["ns", "us", "ms", "s", "m", "h", "D"]
+            units = [
+                "ns",
+                "us",
+                "ms",
+                "s",
+            ]
             res_unit = self.unit
             if hasattr(offset, "offset"):
-                offset_unit = Timedelta(offset.offset).unit
-                idx_self = units.index(self.unit)
-                idx_offset = units.index(offset_unit)
-                res_unit = units[min(idx_self, idx_offset)]
-            dtype = tz_to_dtype(self.tz, unit=res_unit)
-            result = type(self)._simple_new(res_values, dtype=dtype)
+                offset_td = Timedelta(offset.offset)
+                offset_unit = offset_td.unit
+                if self.unit in units and offset_unit in units:
+                    idx_self = units.index(self.unit)
+                    idx_offset = units.index(offset_unit)
+                    res_unit = units[min(idx_self, idx_offset)]
+            dtype_naive = np.dtype(f"datetime64[{res_unit}]")
+            if res_values.dtype != dtype_naive:
+                res_values = res_values.astype(dtype_naive)
+            result = type(self)._simple_new(res_values, dtype=dtype_naive)
 
             if offset.normalize:
                 result = result.normalize()
                 result._freq = None
 
-            if self.tz is not None:
+            if (
+                self.tz is not None
+                and getattr(result.dtype, "tz", None) is None
+                and res_unit == "ns"
+            ):
```

Comment:
why does this codnition need tom change?

Reply:
Well result.dtype.tz is always None, so that part of the check is unnecessary. And I suspect the u it part is just wrong.